### ISSUE TITLE
How to use SVGKit with iOS when loading svg from URL

### ISSUE THREAD

**Issue by chigra** [SVGKit/SVGKit#26]
Hi,

I am new in iOS and I am trying to use SVGKit  to render svg files in my app. I want to load the svg files from URL, but I couldn't get that working. I get the following error: 
Thread 1: EXC_BAD_Access (code = 2, address =0x0) 
on the following line: NSLog(@"[%@] SVGKit Parse error: %@", [self class], *error); in the SVGDocument.m file. 
I tried to implement loading a file from URL by using the already existing SVGPad sample, which works perfectly and loads the pictures but not from URL. I used this code to load the pictures from URL: 

NSString _urlString = [NSString stringWithFormat:@"http://upload.wikimedia.org/wikipedia/commons/6/66/Blank_Map-Africa.svg"];
NSURL *url = [NSURL URLWithString:[urlString stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding]];
SVGDocument_ svgDocument = [SVGDocument documentFromURL:url];
self.contentView = [[[SVGView alloc] initWithDocument:svgDocument] autorelease];

Also I tried to create and my simple app that loads just ten a simple picture from local file by using the following code: 

```
SecondViewController *vc = [[SecondViewController alloc] init];
NSString* svgFilename = [@"/Users/XCode Projects/SVGLink/Beastie" stringByAppendingPathExtension:@"svg"];
SVGDocument *svgDocument = [SVGDocument documentWithContentsOfFile:svgFilename];
 vc.view = [[SVGView alloc] initWithDocument:svgDocument];
[self.window setRootViewController:vc];
```

All I got is this error: [SVGDocument] DEBUG INFO: set document viewBox = {{0, 0}, {340.961, 377.75}}

Can somebody explain me where do I make the error?

Thanks,
Lily

**Comment by adamgit** [comment]
The error reporting in this version is very poor; I have a major update coming soon that has MUCH better errors.

In the meantime... Have you tried looking at your code in the debugger that's built-in to Xcode?

I suspect that if you did, you'd see this line:

```
  stringByAddingPercentEscapesUsingEncoding
```

is breaking your URL, you are passing in an illegal URL if you do that.

**Comment by chigra** [comment]
Last Friday I've found one version of SVGKit, that I can't find it again, where the links worked perfectly and .svg file was loaded correctly. 
In this version, where I am posting this comment, I constantly get the parser error. I guess it is not the link it is my .svg file I've used. 
The same .svg file is shown perfectly in UIWebView but when it comes to SVG, with the working version I've mention before, showed me errors saying that the script and other things can't be parsed. 
So I am guessing that .svg file is the problem, not the link itself. 

I am looking forward to the new version of SVGKit. 

Edited: I think I used this one where the url loading of svg file worked: https://github.com/meric/SVGKit
